proxy/envoy/config.go

```diff
@@ -228,10 +288,15 @@ func buildInboundFilters(instances []*model.ServiceInstance) HTTPRouteConfigs {
 
 			http := httpConfigs.EnsurePort(instance.Endpoint.Port)
 			http.VirtualHosts = append(http.VirtualHosts, host)
+		case model.ProtocolTCP:
+			cluster := buildInboundCluster(instance.Endpoint.Port, port.Protocol)
+			route := buildTCPRoute(cluster, instance.Endpoint.Address, instance.Endpoint.Port)
```

Comment:
buildTCPRoute builds the [L4 routes ](https://lyft.github.io/envoy/docs/configuration/network_filters/tcp_proxy_filter.html#config-network-filters-tcp-proxy-route) which is used later on to contruct the TCP proxy filter.